wallet-new/integration/Main.hs

```diff
@@ -5,115 +5,40 @@ module Main where
 
 import           Universum
 
-import           Cardano.Wallet.Client.Http
-import           Data.Map (fromList)
-import           Data.Traversable (for)
-import           Data.X509.File (readSignedObject)
+import           Cardano.Wallet.Client.Http (WalletClient)
 import           Network.HTTP.Client (Manager)
+import qualified QuickCheckSpecs as QuickCheck
 import           System.Environment (withArgs)
-import           System.IO (hSetEncoding, stdout, utf8)
-import           Test.Hspec
+import           Test.Hspec (Spec, hspec)
 
 import           AccountSpecs (accountSpecs)
 import           AddressSpecs (addressSpecs)
-import           CLI
-import           Functions
+import           CLI (CLIOptions (..), getOptions)
+import           RandomStateWalk (randomStateWalkTest)
+import           SetupTestEnv (setupClient)
+import           System.IO (hSetEncoding, stdout, utf8)
 import           TransactionSpecs (transactionSpecs)
-import           Types
-import           Util (WalletRef, newWalletRef)
+import           Util (WalletRef, newWalletRef, printT)
 import           WalletSpecs (walletSpecs)
 
-import qualified Data.ByteString.Char8 as B8
-import qualified QuickCheckSpecs as QuickCheck
-
-
--- | Here we want to run main when the (local) nodes
--- have started.
+-- | Here we want to run main when the (local) nodes have started.
 main :: IO ()
 main = do
     hSetEncoding stdout utf8
-    CLOptions {..} <- getOptions
-
-    -- TODO (akegalj): run server cluster in haskell, instead of using shell scripts
-    -- serverThread <- async (runWalletServer options)
+    options@CLIOptions {..} <- getOptions
 
     printT "Starting the integration testing for wallet."
-
-    let serverId = (serverHost, B8.pack $ show serverPort)
-    caChain <- readSignedObject tlsCACertPath
-    clientCredentials <- orFail =<< credentialLoadX509 tlsClientCertPath tlsPrivKeyPath
-    manager <- newManager $ mkHttpsManagerSettings serverId caChain clientCredentials
-
-    let baseUrl = BaseUrl Https serverHost serverPort mempty
-
-    let walletClient :: MonadIO m => WalletClient m
-        walletClient = withThrottlingRetry
-            . liftClient
-            $ mkHttpClient baseUrl manager
-
+    (walletClient, manager) <- setupClient options
     -- Acquire the initial state for the deterministic tests
     wRef <- newWalletRef
 
-    -- NOTE Our own CLI options interfere with `hspec` which parse them for
-    -- itself when executed, leading to a VERY unclear message:
-    --
-    --     cardano-integration-test: unrecognized option `--tls-ca-cert'
-    --     Try `cardano-integration-test --help' for more information.
-    --
-    -- See also: https://github.com/hspec/hspec/issues/135
-
```

Comment:
This comment is the result of long hours of debugging. Please, do not discard it.

Reply:
Could you explain to me (and more particularly, as part of a commit's message) why this comment is _plain wrong_ and can be removed :) ? From what I can see, we still rely on `hspec` and not `hspecWith` to run the specs, and if things don't interfere, that is precisely because of the use of this `withArgs` below, isn't it ?

Reply:
sorry that was just a typo in comment.. I've edited the comment:
You are 100% right there is no interference because of the use of `withArgs` 
The fix is to use `withArgs` instead of `hspec` and this is what is done in the code..
I don't know why I invented `hspecWith`

Reply:
I've updated the comment in the code

Reply:
Yeah, the comment was precisely about that actually :upside_down_face: 
Funny enough, there's actually a [hspecWith](http://hackage.haskell.org/package/hspec-core-2.5.8/docs/Test-Hspec-Core-Runner.html#v:hspecWith) which would also solve our issue :sweat_smile: ...